src/api/sso/test/row-level-policies.test.js

```diff
@@ -0,0 +1,350 @@
+          display_name: unknowUser.displayName,
+          github_username: unknowUser.githubUsername,
+          github_avatar_url: unknowUser.githubAvatarUrl,
+        },
+        { returning: 'minimal' }
+      );
+
+      // forbidden, users cannot be created on the client
+      expect(status).toBe(403);
+    });
+
+    it('should not allow updating users', async () => {
+      const { status } = await supabase
+        .from('telescope_profiles')
+        .update({ display_name: unknowUser.displayName })
+        .match({ id: hash(users[0].email) });
+
+      // no rows are returned to be updated, therefore 404
+      expect(status).toBe(404);
+    });
+
+    it('should not allow deleting users', async () => {
+      const { status } = await supabase
+        .from('telescope_profiles')
+        .delete()
+        .match({
+          id: hash(users[0].email),
+        });
+
+      expect(status).toBe(200);
+      // no rows are deleted
+      await ensureUsers(users);
+    });
+  });
+
+  describe('"feeds" security policies', () => {
+    // Create feeds for HansLippershey and an unknown user
+    beforeEach(async () => {
+      await createTelescopeUsers([hansLippershey]);
+      await addFeeds(feedList);
+    });
+
+    afterEach(async () => {
+      await cleanFeeds(feedList);
+      await cleanupTelescopeUsers([hansLippershey]);
+    });
+
+    it('should allow reading the feeds list', async () => {
+      const { status, data: feeds } = await supabase.from('feeds').select('*');
+
+      expect(status).toBe(200);
+      expect(feeds.length).toBe(feedList.length);
+    });
+
+    it('should not allow adding new feeds', async () => {
+      const { status } = await supabase.from('feeds').insert(
+        {
+          url: 'http://dev.to/devto',
+          wiki_author_name: 'devto',
+        },
+        { returning: 'minimal' }
+      );
+
+      // forbidden, unauthenticated users can't add new feeds
+      expect(status).toBe(403);
+    });
+
+    it('should not allow updating feeds', async () => {
+      const { status } = await supabase
+        .from('feeds')
+        .update({ html_url: 'http://dev.to/' }, { returning: 'minimal' })
+        .match({ url: feedList[0].url });
+
+      expect(status).toBe(404);
+    });
+
+    it('should not allow deleting feeds', async () => {
+      const { status } = await supabase.from('feeds').delete().match({
+        url: feedList[0].url,
+      });
+      expect(status).toBe(200);
+      await ensureFeeds(feedList);
+    });
+  });
+});
+
+describe('Authenticated users through Seneca SSO', () => {
+  beforeAll(async () => {
+    const { email, firstName, lastName, displayName, githubAvatarUrl } = hansLippershey;
+    const accessToken = createToken(
+      email,
+      firstName,
+      lastName,
+      displayName,
+      ['seneca', 'telescope'],
+      githubAvatarUrl
+    );
+    // overwrite public anon_key with a JWT sent from the server
+    supabase.auth.setAuth(accessToken);
```

Comment:
I would create a new Supabase client per-test to do this, so there isn't any global state.

Reply:
Adding to my idea above, what if you do:

```js
const supabase = (key = ANON_KEY) => createClient(SUPABASE_URL, key);
```

Now you can pass a different key in tests where that matters.